analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TestClassShouldHaveTestMethod.MsTest.cs

```diff
@@ -113,6 +113,74 @@ public static void AfterTests()
         {
         }
     }
+
+    // Reproducers for https://github.com/SonarSource/sonar-dotnet/issues/5394
+
+    [TestClass]
+    public class SetupAttributes4
+    {
+        [TestInitialize]
+        public static void BeforeTests(TestContext context)
+        {
+        }
+    }
```

Comment:
Only `AssemblyInitialize` makes sense in this scenario. When referring to the case reported by user:
* this only makes sense when used as a base class => should be `abstract` and issue then doesn't exist
* the `[TestClass]` attribute should not be present when the file doesn't contain tests. This class is meant to be inherited and the inheriting class should contain the `[TestClass]` and `[TestMethod]` attributes.